notify/impl.go

```diff
@@ -62,6 +66,79 @@ func Build(confs []*config.Receiver, tmpl *template.Template) map[string]Fanout
 
 const contentTypeJSON = "application/json"
 
+// End-users should not be exposed to Go's type system,
```

Comment:
> I don't think dates and such should be plain strings.

Timestamps are currently float64 in Prometheus templates, the alertmanager doesn't expose any dates yet. I can see us exposing some though for alert starting time, so for consistency we'd be looking at float64 and a humanize function there.

> Anyone making use of such features can be expected to be able to deal with the well-defined template system.

I don't think this is a reasonable expectation. Go's templating is designed for use by the same people writing the code, who will obviously already be familiar with Go's type system and have control of the surrounding code to handle anything that comes up.

What we're doing here is different as we're exposing Go's template system to users who in general will have no knowledge of Go, and may not even know what a type system is. I think losing a handful of minor utility functions is a small price to make things more easily understandable by our whole userbase.